lib/connextclient/ConnextClient.ts

```diff
@@ -0,0 +1,491 @@
+    return true;
+  }
+
+  public getHeight = async () => {
+    return 1; // connext's API does not tell us the height
+  }
+
+  public getConnextInfo = async (): Promise<ConnextInfo> => {
+    let channels: ConnextChannelCount | undefined;
+    let address: string | undefined;
+    let version: string | undefined;
+    let status = 'Ready';
+    const chain =
+      Object.keys(this.tokenAddresses).find(
+        key => this.tokenAddresses.get(key) === this.address,
+      ) || 'connext';
+    if (this.isDisabled()) {
+      status = errors.CONNEXT_IS_DISABLED.message;
+    } else {
+      try {
+        version = (await this.getVersion()).version;
+        const connextChannels = await this.getChannels();
+        channels = {
+          active: connextChannels.filter(c => c.state === 'opened').length,
+          settled: connextChannels.filter(c => c.state === 'settled').length,
+          closed: connextChannels.filter(c => c.state === 'closed').length,
+        };
+        address = this.address;
+        if (channels.active <= 0) {
+          status = errors.CONNEXT_HAS_NO_ACTIVE_CHANNELS().message;
+        }
+      } catch (err) {
+        status = err.message;
+      }
+    }
+
+    return {
+      chain,
+      status,
+      channels,
+      address,
+      version,
+    };
+  }
+
+  /**
+   * Gets the connext version.
+   */
+  public getVersion = async (): Promise<ConnextVersion> => {
+    return ({} as any);
+  }
+
+  /**
+   * Gets info about a given connext payment client.
+   * @param token_address the token address for the network to which the client belongs
+   * @param channel_address the address of the client to query
+   */
+  public getChannel = async (
+    token_address: string,
+    channel_address: string,
+  ): Promise<Channel> => {
+    return ({} as any);
+  }
+
+  /**
+   * Gets info about all non-settled channels.
+   * @param token_address an optional parameter to specify channels belonging to the specified token network
+   */
+  public getChannels = async (token_address?: string): Promise<Channel[]> => {
+    return ({} as any);
+  }
+
+  public channelBalance = async (
+    currency?: string,
+  ): Promise<ChannelBalance> => {
+    const client = this.getConnextClient();
+    if (!currency) {
+      return { balance: 0, pendingOpenBalance: 0, inactiveBalance: 0 };
+    }
+
+    const tokenAddress = this.getTokenAddress(currency);
+
+    const freeBalance = await client.getFreeBalance(tokenAddress);
+
+    const pendingOpenBalance = this.unitConverter.unitsToAmount({
+      currency,
+      units: freeBalance[client.multisigAddress].toNumber(),
+    });
+
+    const inactiveBalance = 0;
+
+    const balance = pendingOpenBalance + inactiveBalance;
+
+    return {
+      balance,
+      pendingOpenBalance,
+      inactiveBalance,
+    };
+  }
+
+  public tradingLimits = async (): Promise<TradingLimits> => {
+    // assume MAX_AMOUNT since Connext will re-collaterize accordingly
```

Comment:
Can you explain a bit here what you mean by re-collaterizing? Eventually we'll want a way to know what our capacity is for sending and receiving payments with connext.